pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -105,6 +113,62 @@ func (f *framework) RunReservePlugins(
 	return nil
 }
 
+func (f *framework) RunPermitPlugins(
+	pc *PluginContext, pod *v1.Pod, nodeName string) *Status {
+	timeout := maxTimeout
+	statusCode := Success
+	for _, pl := range f.permitPlugins {
+		status, d := pl.Permit(pc, pod, nodeName)
+		if !status.IsSuccess() {
+			if status.Code() == Unschedulable {
+				msg := fmt.Sprintf("rejected by %v at permit: %v", pl.Name(), status.Message())
+				klog.V(4).Infof(msg)
+				return NewStatus(status.Code(), msg)
+			}
+			if status.Code() == Wait {
+				if timeout > d {
+					timeout = d
+				}
+				statusCode = Wait
+			} else {
+				msg := fmt.Sprintf("error while running %v permit plugin for pod %v: %v", pl.Name(), pod.Name, status.Message())
+				klog.Error(msg)
+				return NewStatus(Error, msg)
+			}
+		}
+	}
+
+	w := &WaitingPod{
+		pod: pod,
+		s:   make(chan *Status),
+	}
+	f.waitingPods = append(f.waitingPods, w)
```

Comment:
IIUC, when all premit plugins return "Success", we still add the pod to waitingPods. This does not seem correct to me.